server/src/main/java/io/crate/replication/logical/action/PublicationsStateAction.java

```diff
@@ -0,0 +1,274 @@
+                    }
+                }
+            }
+            var response = new Response(
+                relationDetails.stream().map(d -> d.relationName).toList(),
+                relationDetails.stream().flatMap(d -> d.concreteIndices.stream()).toList(),
+                relationDetails.stream().flatMap(d -> d.concreteTemplates.stream()).toList()
+            );
+            listener.onResponse(response);
+        }
+
+        @Override
+        protected ClusterBlockException checkBlock(Request request,
+                                                   ClusterState state) {
+            return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_READ);
+        }
+
+        @Nullable
+        private RelationDetails relationDetail(RelationName relationName, ClusterState clusterState) {
+            var relationDetails = new RelationDetails(relationName);
+            var indexNameOrTemplateName = relationName.indexNameOrAlias();
+            if (clusterState.metadata().hasIndex(indexNameOrTemplateName)) {
+                relationDetails.concreteIndices.add(indexNameOrTemplateName);
+            } else {
+                var templateName = PartitionName.templateName(relationName.schema(), relationName.name());
+                var indexTemplate = clusterState.metadata().templates().get(templateName);
+                if (indexTemplate == null) {
+                    return null;
+                }
+                relationDetails.concreteTemplates.add(templateName);
+                var partitionIndices = Arrays.asList(indexNameExpressionResolver.concreteIndices(
+                    clusterState, IndicesOptions.lenientExpandOpen(), indexNameOrTemplateName));
+                partitionIndices.forEach(i -> relationDetails.concreteIndices.add(i.getName()));
+            }
+            return relationDetails;
+        }
+
+        @VisibleForTesting
+        static List<RelationName> resolveRelationsNames(Publication publication, Schemas schemas) {
+            List<RelationName> relationNames;
+            if (publication.isForAllTables()) {
+                relationNames = InformationSchemaIterables.tablesStream(schemas)
+                    .filter(t -> {
+                        if (t instanceof DocTableInfo dt) {
+                            boolean softDeletes;
+                            if ((softDeletes = IndexSettings.INDEX_SOFT_DELETES_SETTING.get(dt.parameters())) == false) {
+                                LOGGER.warn(
+                                    "Table '{}' won't be replicated as the required table setting " +
+                                    "'soft_deletes.enabled' is set to: {}",
+                                    dt.ident(),
+                                    softDeletes
+                                );
+                                return false;
+                            }
+                            return true;
+                        }
```

Comment:
Would it make sense to check this when the user creates a publication and turn it into a failure?

Reply:
Yes it's checked on publication creation, this path is for new tables included for possible `FOR ALL TABLES` publications.